CMakeLists.txt

```
@@ -71,11 +77,17 @@ if(WIN32)
     message(STATUS "Detected Windows, falling back to using make: ${MAKE_EXECUTABLE}")
 
     set(MSYSTEM "UCRT64")
-    find_file(MSYS2_EXECUTABLE NAMES msys2_shell.cmd REQUIRED HINTS D:/a/_temp/msys64 C:/msys64)
+    set(MSYS2_ARG "-ucrt64")
+    if(CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "ARM64")
```

```suggestion
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "ARM64")
```

For cross compiling (not sure if that will be possible) I think this is probably better

Edit: Oh, looks like you answered that in the readme, but probably still go with this change as it will match the host architecture when not cross compiling.

---

I think no way to compile arm64 version on x64 windows, but i'm not sure arm64 windows build x64 version

---

Anyway, I don't recommend compiling x64 packages under arm64, the ucrt64 component is x64 native binary, and the binary translation effect is very bad.

---

Done. Also add CLANGARM64 check, on who want build win-arm64 target on non-arm64 platform, it will throw a fatal error now.